src/sentry/release_health/metrics.py

```diff
@@ -1486,358 +1431,53 @@ def get_project_releases_count(
             stats_period = "24h"
 
         granularity, stats_start, _ = get_rollup_starts_and_buckets(stats_period, now=now)
-        where = [
-            Condition(Column("timestamp"), Op.GTE, stats_start),
-            Condition(Column("timestamp"), Op.LT, now),
-            Condition(Column("project_id"), Op.IN, project_ids),
-            Condition(Column("org_id"), Op.EQ, organization_id),
-        ]
 
-        try:
-            release_column_name = resolve_tag_key(USE_CASE_ID, organization_id, "release")
-        except MetricIndexNotFound:
-            return 0
+        where = []
 
         if environments is not None:
-            try:
-                environment_column_name = resolve_tag_key(
-                    USE_CASE_ID, organization_id, "environment"
-                )
-            except MetricIndexNotFound:
-                return 0
+            where.append(Condition(Column("tags[environment]"), Op.IN, environments))
 
-            environment_values = resolve_many_weak(USE_CASE_ID, organization_id, environments)
-            where.append(Condition(Column(environment_column_name), Op.IN, environment_values))
-
-        having = []
-
-        # Filter out releases with zero users when sorting by either `users` or `crash_free_users`
-        if scope in ["users", "crash_free_users"]:
-            having.append(Condition(Function("uniq", [Column("value")], "value"), Op.GT, 0))
-            match = Entity(EntityKey.MetricsSets.value)
-            mri = SessionMRI.USER
-        else:
-            match = Entity(EntityKey.MetricsCounters.value)
-            mri = SessionMRI.SESSION
-
-        metric_id = resolve(USE_CASE_ID, organization_id, mri.value)
-        where.append(Condition(Column("metric_id"), Op.EQ, metric_id))
-
-        query_columns = [
-            Function(
-                "uniqExact", [Column(release_column_name), Column("project_id")], alias="count"
-            )
+        if scope == "users":
+            select = [MetricField(metric_mri=SessionMRI.ALL_USER.value, alias="v", op=None)]
+            # where += Condition(Column("value"), Op.GT, 0)
+        elif scope == "crash_free_users":
+            select = [MetricField(metric_mri=SessionMRI.CRASH_FREE_USER.value, alias="v", op=None)]
+            # where += Condition(Column("value"), Op.GT, 0)
+        else:  # sessions
```

Comment:
I assume we will need the `having` clause supported by the Metrics Layer in order for this query to be equivalent when it comes to handling `0` results.

Reply:
Hmm ... not sure why the condition is commented out (that's beside of your Having comment which applies).
It might have remained commented out from some debug experiment ... I'll double check with the original implementation (good catch).

Reply:
Yes, checked, in the original code it was a having clause but it is not available in the Metric layer.
I introduced a post query check for positive results.